src/PostgREST/ApiRequest.hs

```diff
@@ -130,10 +131,12 @@ userApiRequest schema rootSpec req reqBody
       , iAccepts = maybe [CTAny] (map decodeContentType . parseHttpAccept) $ lookupHeader "accept"
       , iPayload = relevantPayload
       , iPreferRepresentation = representation
-      , iPreferSingleObjectParameter = singleObject
+      , iPreferParameters = if hasPrefer (show SingleObject)    then Just SingleObject else
+                            if hasPrefer (show MultipleObjects) then Just MultipleObjects
+                            else Nothing
       , iPreferCount = hasPrefer "count=exact"
-      , iPreferResolution = if hasPrefer (show MergeDuplicates) then Just MergeDuplicates
-                            else if hasPrefer (show IgnoreDuplicates) then Just IgnoreDuplicates
+      , iPreferResolution = if hasPrefer (show MergeDuplicates)  then Just MergeDuplicates else
+                            if hasPrefer (show IgnoreDuplicates) then Just IgnoreDuplicates
```

Comment:
nitpick : wouldn't guards be better for iPreferParameters and iPreferResolution?
```
iPreferResolution
  | hasPrefer (show MergeDuplicates) = Just MergeDuplicates
  | hasPrefer (show IgnoreDuplicates) = Just IgnoreDuplicates
  | otherwise = Nothing
```

Reply:
I see what you mean. Unfortunately GHC doesn't allow using guards in [record syntax](https://en.wikibooks.org/wiki/Haskell/More_on_datatypes#Named_Fields_.28Record_Syntax.29) constructor fields.

I wanted to keep the logic close to the named field since it's short and that way avoid having an extra variable.

Another choice could be using [MultiWayIf](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#extension-MultiWayIf). It would look like:

```haskell
iPreferResolution = 
  if | hasPrefer (show MergeDuplicates)  -> Just MergeDuplicates
     | hasPrefer (show IgnoreDuplicates) -> Just IgnoreDuplicates
     | otherwise                         -> Nothing
```

What do you think, does that make it clearer? Or it'd be better to just have another variable and use regular guards?

Reply:
I was not aware of that limitation of the record syntax.

I do think the MultiWayIf is clearer and I would prefer it over the if-then-else syntax.